### DISCUSSION TITLE
6S lookup tables

### DISCUSSION THREAD

**Discussion by jgrss** [jgrss/geowombat#67] [Q&A] [answered]
@mmann1123 I created a module (see branch `sixs-lut`) for the 6S radiative transfer model with lookup tables for Landsat 5, 7, and 8 and Sentinel 2A and 2B. The files are too large (1-2 GB) to include as package data. Any ideas on how or where to host them so that they are only downloaded when needed?

**Comment by mmann1123** [comment, marked as answer]
Awesome! Happy to sponsor a S3 bucket or dropbox app.  Just let me know what you would prefer.

**Reply by jgrss** [reply]
I would prefer S3 over Dropbox.

**Comment by mmann1123** [comment]
Sure if you send the files I can setup a public bucket.

**Reply by jgrss** [reply]
Thanks for uploading those files @mmann1123. I have a downloader in `geowombat.radiometry.sixs` that looks like:

```python
@staticmethod
def _load(sensor, wavelength, interp_method, from_toar=False):

    if from_toar:
        raise NotImplementedError('Lookup tables from top of atmosphere reflectance are not supported.')
        lut_path = SENSOR_LOOKUP[sensor].path / f'{sensor}_{wavelength}_from_toar.lut'
    else:
        lut_path = SENSOR_LOOKUP[sensor].path / f'{sensor}_{wavelength}.lut'

    if not lut_path.is_file():

        logger.info(f'  Downloading {lut_path.name} into {SENSOR_LOOKUP[sensor].path}.')

        lutd = LUTDownloader()

        lutd.download(f'https://s3geowombat.s3.amazonaws.com/{sensor}_{wavelength}.lut',
                      str(lut_path),
                      safe_download=False)

    lut_ = joblib.load(str(lut_path))

    return lut_[interp_method]
```

Lookup tables will be stored in `~/.geowombat/datasets/lut`.

**Comment by mmann1123** [comment]
Great. Happy to help.
